scripts/test.py

```diff
@@ -2,60 +2,131 @@
+            for root, _, files in os.walk(dirname):
+                for file in files:
+                    if file.split('.')[0] == basename.split('.')[0] and file.endswith('.cpp'):
+                        auxfiles.append(os.path.normpath(os.path.join(root, file)))
+            examples = [] # 寻找样例文件
+            for root, _, files in os.walk(dirname.replace('code', 'examples')):
+                for file in files:
+                    if file.split('.')[0] == basename.split('.')[0] and file.endswith('.in') and os.path.exists(os.path.join(root, file.replace('.in', '.ans'))):
+                        examples.append(os.path.normpath(os.path.join(root, file)))
+
+        elif extname.endswith(('.in', '.ans')): # 单一测试点改变
+            mainfile = os.path.normpath(os.path.join(dirname.replace('examples', 'code'), basename.split('.')[0] + '.cpp'))
+            if mainfile in mainfiles_to_test:
+                continue
+            else:
+                mainfiles_to_test.add(mainfile)
+            auxfiles = [] # 寻找其他辅助文件
+            for root, _, files in os.walk(dirname.replace('examples', 'code')):
+                for file in files:
+                    if file.split('.')[0] == basename.split('.')[0] and file.endswith('.cpp'):
+                        auxfiles.append(os.path.normpath(os.path.join(root, file)))
+            examples = [os.path.normpath(os.path.join(dirname, basename + '.in'))]
+
+        if test(mainfile, auxfiles, examples):
+            cnt_ac += 1
+        else:
+            cnt_error += 1
+        print()
+
 
-    # 判断答案
-    cmd = 'diff -b -B '+outdata+' '+ansdata
-    if os.system(cmd) == 0:
-        print(cpp+' Successfully passed the test')
+    with open(os.environ.get('GITHUB_STEP_SUMMARY'), 'w') as f:
+        f.write(f'# TOTAL {len(mainfiles_to_test)} TESTS, {cnt_ac} ACCEPTED, {cnt_skip} SKIPPED, {cnt_error} ERROR\n\n')
+        f.write(summary)
+    print(f'::group::TOTAL {len(mainfiles_to_test)} TESTS, {cnt_ac} ACCEPTED, {cnt_skip} SKIPPED, {cnt_error} ERROR\n::endgroup::')
+    if cnt_error:
+        sys.exit(1)
+
+
+def test(mainfile, auxfiles, examples) -> bool:
+    global summary
+    print(f'::group::Test for {mainfile}...')
+    # 检测文件存在
+    for file in auxfiles:
+        if not os.path.exists(file):
+            print(f'::endgroup::')
+            print(f'::error file={file},title=file {file} not found::')
+            summary += f'## 找不到文件：{file}\n对{mainfile}的测试因找不到文件{file}而被迫中止\n\n'
+            return False
+    for file in examples:
+        if not os.path.exists(file):
+            print(f'::endgroup::')
+            print(f'::error file={file},title=file {file} not found::')
+            summary += f'## 找不到文件：{file}\n对{mainfile}的测试因找不到文件{file}而被迫中止\n\n'
+            return False
+        
+    # 编译
+    compile_command = f'g++ -std=c++17 {" ".join(auxfiles)} -o {mainfile.split(".")[0]}'
+    print(compile_command, end=' ')
+
+    if os.system(compile_command) != 0:
```

Comment:
Use `subprocess` module instead. Same below.
https://docs.python.org/3/library/subprocess.html

Reply:
Fixed 👋 GitHub Action record is https://github.com/OI-wiki/OI-wiki/actions/runs/10092500442